app/models/concerns/taxonomix.rb

```diff
@@ -132,6 +132,44 @@ def scope_by_taxable_ids(scope)
         scope.where("#{table_name}.id IN (#{cached_ids.join(',')})")
       end
     end
+
+    def join_user_organizations
+      join_taxonomies(user_organizations, org_join_arel)
+    end
+
+    def join_user_locations
+      join_taxonomies(user_locations, loc_join_arel)
+    end
+
+    def join_taxonomies(taxonomies, tax_arel)
+      class_arel = base_class.arel_table
+      statement = class_arel.join(tax_arel, Arel::Nodes::OuterJoin).on(
```

Comment:
This is specific to the audits case, where you need the outer join so you can also see the partially taxed and untaxed resources. in other cases the outer join doesn't make sense. 
Also, not all models actually include taxonomix - specifically those that are not fully taxed.
Should these methods all live inside the audits extension and not included in all models using taxonomix?

Reply:
I think the outer joins make sense for all models that include taxonomix and even partially taxed but it is only efficient if the models records can grow large like Audit. It is because outer joins is a little bit slower for smaller tables comparing to the current way. For example, ForemanOpenscap::ArfReport might be the candidate that can apply the outer join..

Reply:
I think Tomer is comparing OUTER vs INNER joins. And I'd agree that if we use INNER join, we can use the join like the only mean to filter the records, we wouldn't need further conditions on the joined table.

Reply:
OTOH, outer join is more flexible. BUT then it doesn't have much sense to use the `taxonomy_ids` in the join itself, because we need another condition using the ids.